modules/internal/ChapelBase.chpl

```diff
@@ -307,11 +307,31 @@ module ChapelBase {
   //
   // % on primitive types
   //
-  inline proc %(a: int(?w), b: int(w)) return __primitive("%", a, b);
-  inline proc %(a: uint(?w), b: uint(w)) return __primitive("%", a, b);
+  inline proc %(a: int(?w), b: int(w)) {
+    if (chpl_checkDivByZero) then
+      if b == 0 then
+        halt("Attempt to divide by zero");
```

Comment:
I'm not sure of the answer to this question:  Is "divide" an appropriate term here when it's modulus rather than a divide that we're computing?  If there's precedent in other systems, it's fine with me.